### ISSUE TITLE
Unique body-parts are good in compiled training set, but are blank in analysis results of testing videos

### ISSUE THREAD

**Issue by Xiaoyu-Tong** [DeepLabCut/DeepLabCut#952]
Thanks for opening this issue, and thanks for using DeepLabCut (we hope you are enjoying it ;). Please fill out the template completely, including the full "traceback" and input code that you ran to hit this error.

**Your Operating system and DeepLabCut version**

I was using Google Colab to run model training and video analysis of a multianimal DLC project. And I used DLC version 2.2b7.

**Describe the problem**

I was training and testing a maDLC model with an unique body-part. But the unique body-part was not predicted for testing videos. And I have checked the compiled training set. The data of unique body-part was good.

```
deeplabcut.create_multianimaltraining_dataset(path_config_file, windows2linux=True)
shuffle = 1
deeplabcut.train_network(path_config_file, shuffle=shuffle, displayiters=1000,saveiters=4000)
videofile_path = ['/content/drive/My Drive/test.mp4'] 
scorername = deeplabcut.analyze_videos(path_config_file, 
                                       videofile_path, 
                                       shuffle=1, 
                                       videotype=VideoType, 
                                       c_engine=False)
tracktype= 'skeleton' 
deeplabcut.convert_detections2tracklets(path_config_file, Specific_videofile, videotype=VideoType,
                                                    shuffle=1, track_method=tracktype, overwrite=True)
deeplabcut.convert_raw_tracks_to_h5(path_config_file, picklefile)
deeplabcut.filterpredictions(path_config_file, 
                                 videofile_path, 
                                 videotype='mp4', 
                                 track_method = 'skeleton')
```

I got no error message. And I could generate labeled videos from results, but videos were without labels of the unique body-part.

**How to Reproduce the problem**
1. Run the above code
2. Check resultant csv files of testing videos (and to check csv file of compiled training set to make sure dataset creation is good).

**Screenshots**
Config file: (just want to make sure the format is correct)
![image](https://user-images.githubusercontent.com/55998514/95656694-081bf700-0ade-11eb-88a2-4c41ba733c6f.png)

Unique bodypart column in compiled training set (the unique bp is called implant):
![image](https://user-images.githubusercontent.com/55998514/95656739-57622780-0ade-11eb-8556-797d43106e44.png)

Unique bodypart column in csv file of a testing video:
![image](https://user-images.githubusercontent.com/55998514/95656778-998b6900-0ade-11eb-9fb3-62e67bf7b7df.png)

**Additional context**

**Comment by MMathisLab** [comment]
are **all** the point there when you run: 

```python
scorername = deeplabcut.analyze_videos(config_path,['/fullpath/project/videos/testVideo.mp4'], videotype='.mp4')
deeplabcut.create_video_with_all_detections(config_path, ['/fullpath/project/videos/testVideo.mp4'], scorername)
```

see here: https://github.com/DeepLabCut/DeepLabCut/blob/master/docs/maDLC_AdvUserGuide.md#attention

**Comment by jeylau** [comment]
Hey @Xiaoyu-Tong, could you retry by passing `min_tracklet_len=0` to `convert_raw_tracks_to_h5`? That should solve the issue. That's a behavior I did not anticipate though, I'll push a fix tomorrow!

**Comment by Xiaoyu-Tong** [comment]
Hey @jeylau , thanks for the update. I have tried min_tracklet_len=0 but there was an issue which prevented it from working. When I ran convert_raw_tracks_to_h5 without min_tracklet_len=0 , everything was alright except columns for the unique bodypart was blank. However, when I ran convert_raw_tracks_to_h5 with min_tracklet_len=0, I always got a MemoryError. I have tried it both on Google Colab and on my desktop. On google Colab, it said the session crashed after using all available RAMs. I had encountered this issue before, and was able to solve it by upgrading my account to premium (RAM from 12.5Gb to 25Gb). But this time further upgrade of RAMs is not possible. Then I tried it on my local computer, and got a MemoryError (complete messages below):
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\ProgramData\Anaconda3\envs\dlc-CPU\lib\site-packages\deeplabcut\refine_training_dataset\tracklets.py", line 1134, in convert_raw_tracks_to_h5
    manager.load_tracklets_from_pickle(tracks_pickle)
  File "C:\ProgramData\Anaconda3\envs\dlc-CPU\lib\site-packages\deeplabcut\refine_training_dataset\tracklets.py", line 332, in load_tracklets_from_pickle
    self._load_tracklets(tracklets, auto_fill)
  File "C:\ProgramData\Anaconda3\envs\dlc-CPU\lib\site-packages\deeplabcut\refine_training_dataset\tracklets.py", line 205, in _load_tracklets
    is_single = np.isnan(to_fill[:, mask_multi]).all()
MemoryError

*****************
So basically I could not get any results from convert_raw_tracks_to_h5 this time. As I already got 25Gb RAMs on Google Colab, and the pickle file is only 26Mb, I feel like there is something unexpected here... Do you have any idea about this? Thank you!

**Comment by jeylau** [comment]
Please retry with 2.2b8, memory management has drastically improved. Otherwise, wait a little longer for b9, the bug you described will be fixed 😃
